### ISSUE TITLE
[UI] async jobs from UI cannot be executed in parallel safely 

### ISSUE THREAD

**Issue by DaanHoogland** [apache/cloudstack#5883]
When you execute for instance a delete and than another delete, you might end up deleting the worng item. This happens when the first delete completes and teh second is fired a fragment of a second later. The item choosen to delete will default to item 0 after refresh. If another item was intended this is not honoured and the delete is executed regardless.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~
create ten  VMs
delete them one by one from the bottom, by using the quick actions menus
before you hit the top at some point the top one will have been deleted before the ones below
~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by utchoang** [comment]
@DaanHoogland I have checked and it looks like the results returned after the API query are mixed compared to the original time. I think the problem comes from ordering in the returned result and not from the UI

**Comment by DaanHoogland** [comment]
> @DaanHoogland I have checked and it looks like the results returned after the API query are mixed compared to the original time. I think the problem comes from ordering in the returned result and not from the UI

Sorry, but Ii cannot accept that argument @utchoang. The delete is based on UUID. The user doesn't see that UUID. If the UI decides to delete another item, it is because of a race condition in the UI. It should not be based on a list other than what the user saw when issuing the command.

**Comment by utchoang** [comment]
@DaanHoogland Sorry, maybe I'm misunderstanding what you mean. I will test it again following the steps.

**Comment by utchoang** [comment]
@DaanHoogland It is true that I misunderstood your problem. This issue can occur for other similar actions that use custom components.

**Comment by sureshanaparti** [comment]
Fixed in #5884
